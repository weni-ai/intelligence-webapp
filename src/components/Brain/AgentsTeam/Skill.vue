<template>
  <section class="skill">
    <p
      class="skill__icon"
      data-testid="skill-icon"
    >
      {{ icon }}
    </p>

    <UnnnicIntelligenceText
      class="skill__title"
      tag="p"
      family="secondary"
      size="body-md"
      color="neutral-cloudy"
      data-testid="skill-name"
    >
      {{ title }}
    </UnnnicIntelligenceText>
  </section>
</template>

<script setup>
const props = defineProps({
  title: {
    type: String,
    required: true,
  },
  icon: {
    type: String,
    required: true,
  },
});
</script>

<style lang="scss" scoped>
.skill {
  border-radius: $unnnic-border-radius-sm;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;

  padding: $unnnic-spacing-nano $unnnic-spacing-xs;

  display: flex;
  align-items: center;
  gap: $unnnic-spacing-nano;

  &__icon {
    font-size: $unnnic-icon-size-xs;
    line-height: 0;
  }

  &__title {
    white-space: nowrap;
  }
}
</style>
