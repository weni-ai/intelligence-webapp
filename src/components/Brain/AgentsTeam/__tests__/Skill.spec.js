import { mount } from '@vue/test-utils';
import Skill from '../Skill.vue';

describe('Skill.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Skill, {
      props: {
        title: 'Test Title',
        icon: 'icon-1',
      },
    });
  });
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders icon correctly', () => {
    const icon = wrapper.find('[data-testid="skill-icon"]');
    expect(icon.exists()).toBe(true);
    expect(icon.text()).toBe('icon-1');
  });

  it('renders title correctly', () => {
    const title = wrapper.find('[data-testid="skill-name"]');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe('Test Title');
  });
});
