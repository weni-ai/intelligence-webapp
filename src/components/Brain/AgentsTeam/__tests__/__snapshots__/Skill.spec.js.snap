// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skill.vue > renders correctly 1`] = `
<section
  class="skill"
  data-v-29c532b5=""
>
  <p
    class="skill__icon"
    data-testid="skill-icon"
    data-v-29c532b5=""
  >
    icon-1
  </p>
  <p
    class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy skill__title skill__title"
    data-testid="skill-name"
    data-v-29c532b5=""
    data-v-ec8da4ad=""
  >
    
    Test Title
    
  </p>
</section>
`;
